frontend/public/components/autocomplete.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { TextInput } from '@patternfly/react-core';
+import { useDocumentListener, getLabelsAsString } from '@console/shared';
+import { KeyEventModes } from '@console/shared/src/hooks';
+import { fuzzyCaseInsensitive } from './factory/table-filters';
+import { K8sResourceCommon } from '../module/k8s';
+
+const MAX_SUGGESTIONS = 5;
+
+const labelParser = (resources: K8sResourceCommon[]) => {
```

Comment:
nit: I'd be explicit about the return type here (although TypeScript might be able to infer).

```suggestion
const labelParser = (resources: K8sResourceCommon[]): Set<string> => {
```